railties/lib/rails/app_version.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module Rails
+  # Manages the application revision for deployment tracking and error reporting.
+  #
+  # The revision is determined in order of precedence:
+  # 1. Custom proc/string via config.revision
+  # 2. REVISION file in application root (created by deployment tools like Capistrano)
+  # 3. nil if none available
+  module AppVersion
+    class << self
+      def revision(config_revision)
+        @revision ||= load_revision(config_revision)
```

Comment:
This means that if it's not configured, we'll keep trying to check for the `REVISION` file. 

Given it's likely we'll expose the revision in various places (typically `Rails.error.report`) we should be able to memoize `nil`.